src/connection.js

```diff
@@ -262,6 +267,10 @@ class Connection extends EventEmitter {
       }
     }
 
+    if (this.config.domain && !this.config.userName && !this.config.password && SspiModuleSupported) {
+      this.config.options.useWindowsIntegratedAuth = true;
```

Comment:
If userName and password is supplied along with connection property that enables integrated auth, it would be good to ignore the userName/password and connect with integrated auth, as its more secure. 

```node
if (this.config.IntegratedSecurity && SspiModuleSupported) {
this.config.options.useWindowsIntegratedAuth = true; }
```

This is usually the behavior in other client drivers.